tools/tidy/include/ASTHelperVisitors.h

```diff
@@ -35,10 +36,25 @@ struct TidyVisitor {
         auto parentPath = weakly_canonical(std::filesystem::path(path).parent_path());
         const auto& skipFiles = config.getSkipFiles();
         const auto& skipPaths = config.getSkipPaths();
-        return std::find(skipFiles.begin(), skipFiles.end(), file) != skipFiles.end() ||
-               std::find_if(skipPaths.begin(), skipPaths.end(), [&](auto& path) {
-                   return parentPath.string().find(path) != std::string::npos;
-               }) != skipPaths.end();
+        const auto& skipPatterns = config.getSkipPatterns();
+
+        // Check skip files and skip paths (existing logic)
+        if (std::find(skipFiles.begin(), skipFiles.end(), file) != skipFiles.end() ||
```

Comment:
You can remove all of the logic for skip files and skip paths. The new skip patterns feature can accomplish all of the same functionality and more.